src/bacnet/basic/tsm/tsm.h

```diff
@@ -21,8 +21,8 @@
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 *********************************************************************/
-#ifndef TSM_H
-#define TSM_H
+#ifndef BACNET_BASIC_TSM_TSM_H
+#define BACNET_BASIC_TSM_TSM_H
```

Comment:
I think that we should start changing those guards when we touch a file as they can cause trouble with legacy apps e.g. 

config.h uses `CONFIG_H`, abort.h uses `ABORT_H` etc. which are probably used also by other old libraries

best would be imho the usage of `#pragma once` as then you never have copy/paste issues

Reply:
If you use a private prefix, then people using that prefix will enter undefined behaviour.

Reply:
why private prefix? this is just a namespaced prefix to not collide with other libraries...

basically the same as https://github.com/bacnet-stack/bacnet-stack/pull/91

Reply:
I meant the same thing. Use a private namespace (= private prefix) in the include guard. (I don't mean c++ namespaces)